packages/venia-ui/lib/components/CartPage/PriceAdjustments/GiftOptions/GiftOptions.js

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useMemo } from 'react';
+
+import useGiftOptions from '@magento/peregrine/lib/talons/CartPage/GiftOptions/useGiftOptions';
+
+import Checkbox from '../../../Checkbox';
+import TextArea from '../../../TextArea';
+import { mergeClasses } from '../../../../classify';
+
+import defaultClasses from './giftOptions.css';
+
+const GiftOptions = props => {
+    const [
+        { includeGiftReceipt, includePrintedCard, giftMessage },
+        {
+            toggleIncludeGiftReceiptFlag,
+            toggleIncludePrintedCardFlag,
+            updateGiftMessage
+        }
+    ] = useGiftOptions();
+
+    const setGiftMessage = useCallback(
```

Comment:
I wasn't sure if I should be having hooks in the view layer. Thought it made more sense here than in the talon. Let me know if this needs to change.

Reply:
Yep - only UI code lives in components. Logic lives in talons. move callbacks into the talon.